manifests/certonly.pp

```diff
@@ -67,18 +67,35 @@
     $command_start = "${letsencrypt_command} --text --agree-tos --non-interactive certonly -a ${plugin} "
   }
 
-  if $plugin == 'webroot' {
-    $_command_domains = zip($domains, $webroot_paths).map |$domain| {
-      if $domain[1] {
-        "--webroot-path ${domain[1]} -d ${domain[0]}"
-      } else {
-        "-d ${domain[0]}"
+  case $plugin {
+
+    'webroot': {
+      $_command_domains = zip($domains, $webroot_paths).map |$domain| {
+        if $domain[1] {
+          "--webroot-path ${domain[1]} -d ${domain[0]}"
+        } else {
+          "-d ${domain[0]}"
+        }
       }
+      $command_domains = join([ "--cert-name ${title}", ] + $_command_domains, ' ')
     }
-    $command_domains = join([ "--cert-name ${title}", ] + $_command_domains, ' ')
-  } else {
-    $_command_domains = join($domains, ' -d ')
-    $command_domains  = "--cert-name ${title} -d ${_command_domains}"
+
+    'dns-rfc2136': {
+      require ::letsencrypt::plugin::dns_rfc2136
```

Comment:
please don't add leading `::` in front of class names or variables. That is considered legacy syntax. It was required for Puppet 3 support.

Reply:
Good to know. There are 3 usages of it at the bottom of the file not related to my code. Update those at the same time?

There's also a couple of `${::path}` usages for execs, but those seem to require it to avoid local scope. Curious if that is an acceptable exception?

Reply:
Hrm. Those should probably be `$facts['path']` instead.

Reply:
correct. For facts, `$facts['path']` should be used.